testutils/file_utils.go

```diff
@@ -100,33 +103,99 @@ func BuildTempModuleWithFirstRun(tb testing.TB, modDir string) string {
 // MockBuffer is a buffered writer that just appends data to an array to read
 // without needing a real file system for testing.
 type MockBuffer struct {
-	lock   sync.Mutex
-	Writes []*v1.SensorData
+	ctx    context.Context
+	cancel context.CancelFunc
+	Writes chan *v1.SensorData
 }
 
-// Write adds a collected sensor reading to the array.
+// NewMockBuffer returns a mock buffer.
```

Comment:
consider documenting this needs to be `Close()` _before_ the collector closes

Reply:
done